inspirehep/modules/orcid/api.py

```diff
@@ -0,0 +1,238 @@
+            recid,
+            put_code,
+            orcid
+        ))
+        orcid_api.update_record(
+            orcid_id=orcid,
+            token=oauth_token,
+            request_type='work',
+            data=orcid_xml,
+            put_code=put_code,
+            content_type='application/orcid+xml',
+        )
+    else:
+        LOGGER.info(
+            "No put-code found, pushing new record #{} to ORCID {}.".format(
+                recid,
+                orcid,
+            )
+        )
+        put_code = orcid_api.add_record(
+            orcid_id=orcid,
+            token=oauth_token,
+            request_type='work',
+            data=orcid_xml,
+            content_type='application/orcid+xml',
+        )
+
+        LOGGER.info("Record added with put-code {}.".format(put_code))
+
+    return put_code
+
+
+def _get_record_by_mime(config, recid, mime_type):
+    """
+
+    Args:
+        config (inspire_utils.config.Config): configuration
+        recid (string): HEP record ID
+        mime_type (string): accept type
+
+    Returns:
+        requests.Response: response form API
+    """
+    server_name = config['SERVER_NAME']
+
+    record_api_endpoint = _get_api_url_for_recid(
+        server_name, config['SEARCH_UI_SEARCH_API'], recid
+    )
+
+    response = requests.get(record_api_endpoint, headers={
+        'Accept': mime_type
+    })
+    response.raise_for_status()
+    return response
+
+
+def _get_hep_record(config, recid):
+    """
+
+    Args:
+        config (inspire_utils.config.Config): configuration
+        recid (string): HEP record ID
+
+    Returns:
+        dict: HEP record
+    """
+    hep_response = _get_record_by_mime(config, recid, 'application/json')
+    return hep_response.json()['metadata']
+
+
+def _get_bibtex_record(config, recid):
+    """
+
+    Args:
+        config (inspire_utils.config.Config): configuration
+        recid (string): HEP record ID
+
+    Returns:
+        dict: BibTeX serialized record
+    """
+    bibtex_response = _get_record_by_mime(config, recid, 'application/x-bibtex')
+    return bibtex_response.text
+
+
+def get_author_putcodes(orcid, oauth_token):
+    """Get put-codes for author works pushed by INSPIRE.
+
+    Args:
+        orcid (string): author's ORCID
+        oauth_token (string): author's OAUTH token
+
+    Returns:
+        Tuple[
+            List[Tuple[string, string]],
+            List[string]
+        ]: list of tuples of the form (recid, put_code) with results,
+            and another list of put-codes for which we failed to obtain
+            the recids (should never happen)
+    """
+    api = _get_api()
+    # This reads the record _summary_ (no URLs attached):
+    user_works = api.read_record_member(
+        orcid,
+        'works',
+        oauth_token,
+        accept_type='application/orcid+json',
+    )
+
+    put_codes = [
+        str(item[0]) for item
+        in get_value(user_works, 'group.work-summary.put-code')
```

Comment:
This code breaks if `user_works` has no `group.work-summary.put-code`.
I'm not sure if this is even possible (it's ORCID's API data indeed) but maybe this would be safer for us:
`get_value(user_works, 'group.work-summary.put-code', [])`